ext/fluxnet_simulations/data_processing.jl

```diff
@@ -143,6 +157,32 @@ function get_data_at_start_date(
     return preprocess_func(v[idx_start])
 end
 
+"""
+    read_fluxnet_data(site_ID; hour_offset_from_UTC = nothing)
+
+Reads Fluxnet CSV data and processes timestamps, converting local time to UTC.
+Returns (data, columns, local_datetime, UTC_datetime).
+
+If `hour_offset_from_UTC` is nothing, the UTC datetime is not returned.
+"""
+function read_fluxnet_data(site_ID; hour_offset_from_UTC = nothing)
```

Comment:
It kind of seems like this function does two things in one function (reads the data, and returns specific columns and transformations of a column also), but often we only need it to do one of those things (how it used later on, where we overwrite the time vectors).

Maybe it should be split into two:
(data, columns) = read_fluxnet_data(site_ID)
UTC_datetimes = get_UTC_datetimes(data, columns; timestamp = "start")
UTC_datetimes = get_UTC_datetimes(data, columns; timestamp = "end")
UTC_datetimes = get_UTC_datetimes(data, columns; timestamp = "average")

and local datetimes shouldnt be returned?